python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -344,6 +344,38 @@ def groups(self):
             zip(group_names.to_pandas(), grouped_index._split(offsets[1:-1]))
         )
 
+    @cached_property
+    def indices(self):
+        """
+        Dict {group name -> group indices}.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> data = [[10, 20, 30], [10, 30, 40], [40, 50, 30]]
+        >>> df = cudf.DataFrame(data, columns=["a", "b", "c"])
+        >>> df
+            a   b   c
+        0  10  20  30
+        1  10  30  40
+        2  40  50  30
+        >>> df.groupby(by=["a"]).indices
+        {10: array([0, 1]), 40: array([2])}
+        """
+        group_names, offsets, _, grouped_values = self._grouped()
+
+        if len(group_names) > self._MAX_GROUPS_BEFORE_WARN:
+            warnings.warn(
+                f"GroupBy.indices performance scales poorly with "
+                f"number of groups. Got {len(group_names)} groups."
+            )
+        return dict(
+            zip(
+                group_names.to_pandas(),
+                np.split(grouped_values.index.values, offsets[1:-1]),
```

Comment:
**question**: This delivers array pieces that are on device (`grouped_values.index.values` is a cupy array). Do we want host-side values (using `.values_host`)

Reply:
Maybe not for now. I don't see a compelling reason to return `.values_host` here yet.